Added new quickstart docs.

lonnieezell opened the pull request:
Created a new docs page, QuickStart, that provides a quick cheat sheet of some of the items a developer needs to know how to do on a daily basis in order to get up and running as quick as possible with Shield.

MGatner commented on docs/2 - quickstart.md:
```diff
@@ -0,0 +1,260 @@
+
+### Deleting Users
+
+A user's data can be spread over a few different tables so you might be concerned about how to delete all of the user's data from the system. This is handled automatically at the database level for all information that Shield knows about, through the `onCascade` settings of the table's foreign keys. You can delete a user like any other entity.
```
This is a bit misleading since our UserModel implements soft deletes by default. Please either update the example to be a hard delete or differentiate the process more.

MGatner commented on docs/2 - quickstart.md:
```diff
@@ -0,0 +1,260 @@
+
+### Editing A User
+
+The `UserModel::save()` method has been modified to ensure that an email or password previously set on the `User` entity will be automatically in the correct `UserIdentity` record.
```
```suggestion
The `UserModel::save()` method has been modified to ensure that an email or password previously set on the `User` entity will be automatically updated in the correct `UserIdentity` record.
```

MGatner commented on docs/2 - quickstart.md:
````diff
@@ -0,0 +1,260 @@
+
+```php
+$users = model('UserModel');
+$user = $users->findById(123);
````
```suggestion
$user  = $users->findById(123);
```

MGatner commented on docs/2 - quickstart.md:
````diff
@@ -0,0 +1,260 @@
+$users->save($user);
+```
+
+If you prefer to user the `update()` method, than you will have to update the user's appropriate UserIdentity manually.
````
```suggestion
If you prefer to use the `update()` method then you will have to update the user's appropriate UserIdentity manually.
```

MGatner commented on docs/2 - quickstart.md:
```diff
@@ -0,0 +1,260 @@
+]);
+
+// Saves the username field
+$users->save($user);
```
I think you meant for this to be an example of using `update()`, no?
```suggestion
$users->update($user);
```

MGatner reviewed (changes requested):
This is great! Even though I feel like we have very good docs people have still been asking for help getting started. I think this is a strong bridge to that end. 

Mostly editorial comments, but please fix the deletion section particularly.

lonnieezell commented on docs/1 - concepts.md:
````diff
@@ -25,6 +25,7 @@ The only requirement is that your new class MUST extend the provided `UserModel`
 public $userProvider = 'CodeIgniter\Shield\Models\UserModel';
 ```
 
+<a name="identities" />
````
doh! I forgot that. Fixing.

lonnieezell commented on docs/2 - quickstart.md:
```diff
@@ -0,0 +1,260 @@
+
+### Customize Remember-me functionality
+
+Remember-me functionality is enabled by default for the `Session` handler. While this is handled in a secure manner, some sites may want it disbled. You might also want to change how long it remberers a user and doesn't require additional login.
```
You typoed my typo :) 

```suggestion
Remember-me functionality is enabled by default for the `Session` handler. While this is handled in a secure manner, some sites may want it disabled. You might also want to change how long it remembers a user and doesn't require additional login.
```

lonnieezell commented on docs/2 - quickstart.md:
````diff
@@ -0,0 +1,260 @@
+    'register' => \CodeIgniter\Shield\Authentication\Actions\EmailActivator,
+];
+```
+
+
+
````
The line breaks serve as a way to find the sections when scanning through.

lonnieezell commented on docs/2 - quickstart.md:
````diff
@@ -0,0 +1,260 @@
+```php
+$user = auth()->user();
+
+$user->syncPermissions(['users.create', 'beta.access']);
````
Yeah, I saw that. Luckily, we're still in beta. It's on my mental fix-list. :)

lonnieezell commented on docs/2 - quickstart.md:
```diff
@@ -0,0 +1,260 @@
+
+This can also be done through a [controller filter](https://codeigniter.com/user_guide/incoming/filters.html) if you want to apply it to multiple pages of your site.
+
+
+
+
```
These serve as a visual indicator of section breaks when editing the file and scanning the page. It shouldn't cause any extra HTML, I don't believe.

kenjis commented:
The current document structure is not good, because if a new file is added, many URLs of other files will be changed.
The link that was created in the past will be 404.

MGatner commented:
@kenjis Do you mean the numbering? We could make an index.md file with our own ordering and then leave the files themselves unnumbered.

kenjis commented:
@MGatner Yes, it may change filenames again and again. Generally, URLs should not be changed.

lonnieezell commented:
I believe I've addressed all comments. I just pushed changes that removed the numbers from the doc files, added an `_index.md` file, and added a TOC to each docs file.

kenjis commented:
@lonnieezell This branch has conflicts that must be resolved.

kenjis commented:
I don't know.

The TOC is something wrong.
https://github.com/codeigniter4/shield/blob/d49bbb62b79b2bcea95e64f89df27f1d7b9999ea/docs/install.md

datamweb commented:
Friends, I have not yet had the opportunity to read the full **PR**, but at first glance it looks great. Quick installation was required.
**thanks all**
